app/models/ems_refresh/refreshers/ems_refresher_mixin.rb

```diff
@@ -52,8 +45,76 @@ def refresh
         _log.info "Refreshing all targets...Complete"
       end
 
-      def save_inventory(ems, _targets, hashes)
-        EmsRefresh.save_ems_inventory(ems, hashes)
+      def preprocess_targets
+        @full_refresh_threshold = options[:full_refresh_threshold] || 10
+
+        # See if any should be escalated to a full refresh
+        @targets_by_ems_id.each do |ems_id, targets|
+          ems = @ems_by_ems_id[ems_id]
+          ems_in_list = targets.any? { |t| t.kind_of?(ExtManagementSystem) }
+
+          if ems_in_list
+            _log.info "Defaulting to full refresh for EMS: [#{ems.name}], id: [#{ems.id}]." if targets.length > 1
+            targets.clear << ems
+          elsif !ems_in_list && targets.length >= @full_refresh_threshold
+            _log.info "Escalating to full refresh for EMS: [#{ems.name}], id: [#{ems.id}]."
+            targets.clear << ems
+          end
+        end
+      end
+
+      def refresh_targets_for_ems(ems, targets)
+        # handle a 3-part inventory refresh process
+        # 1. collect inventory
+        # 2. parse inventory
+        # 3. save inventory
+        log_header = format_ems_for_logging(ems)
+
+        targets_with_inventory, _ = Benchmark.realtime_block(:get_inventory_for_targets) do
+          get_inventory_for_targets(ems, targets)
```

Comment:
rubocop doesn't like methods starting with `get_`...can we make this `collect_inventory_...`